python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -84,37 +102,6 @@ class ArrowTensorArray(pa.ExtensionArray):
 
     OFFSET_DTYPE = np.int32
 
-    def __getitem__(self, key):
-        # This __getitem__ hook allows us to support proper
-        # indexing when accessing a single tensor (a "scalar" item of the
-        # array). Without this hook for integer keys, the indexing will fail on
-        # all currently released pyarrow versions due to a lack of proper
-        # ExtensionScalar support. Support was added in
-        # https://github.com/apache/arrow/pull/10904, but hasn't been released
-        # at the time of this comment, and even with this support, the returned
-        # ndarray is a flat representation of the n-dimensional tensor.
-
-        # NOTE(Clark): We'd like to override the pa.Array.getitem() helper
-        # instead, which would obviate the need for overriding __iter__()
-        # below, but unfortunately overriding Cython cdef methods with normal
-        # Python methods isn't allowed.
-        if isinstance(key, slice):
-            return super().__getitem__(key)
-        return self._to_numpy(key)
-
-    def __iter__(self):
-        # Override pa.Array.__iter__() in order to return an iterator of
-        # properly shaped tensors instead of an iterator of flattened tensors.
-        # See comment in above __getitem__ method.
-        for i in range(len(self)):
-            # Use overridden __getitem__ method.
-            yield self.__getitem__(i)
-
-    def to_pylist(self):
-        # Override pa.Array.to_pylist() due to a lack of ExtensionScalar
-        # support (see comment in __getitem__).
-        return list(self)
```

Comment:
This is no longer need for Arrow 9+, which has proper `ExtensionScalar` support that we now leverage.